docs/source/keylime/index.rst

```diff
@@ -0,0 +1,97 @@
+Keylime
+=======
+
+ESI provides a guide to use `keylime`_ outside Ironic for run-time attestation with `IMA`_.
+
+Prerequisites
+-------------
+* A Keylime server machine with keylime registrar and verifier installed.
+* A node to be attested with an active TPM 2.0 and keylime agent installed.
+* A private network for attestation workflow.
+
+Build Keylime Server
+--------------------
+Install Keylime packages:
+
+.. prompt:: bash $
+
+  git clone https://github.com/keylime/keylime.git
+  cd keylime && ./installer.sh
+
+Modify /etc/keylime.conf:
+
+* Set ``require_ek_cert`` to false if the physical TPM does not have EK certificates.
+* Set the binding address for verifier and registrar, for example:
+
+  .. prompt::
+
+    [cloud_verifier]
+    cloudverifier_ip = 0.0.0.0
+
+    [registrar]
+    registrar_ip = 0.0.0.0
+
+Start Keylime services:
+
+.. prompt:: bash $
+
+  keylime_verifier
+  keylime_registrar
+
+Setup Keylime Agent
+-------------------
+`Diskimage-builder tool`_ could be used to build provisioning image with keylime-agent installed.
+
+* An example to use diskimage-builder tool to build a CentOS 8 image:
+
+  .. prompt:: bash $
+
+    export DIB_KEYLIME_AGENT_REGISTRAR_IP=<keylime_registrar_service_ip>
+    export DIB_KEYLIME_AGENT_REGISTRAR_PORT=<keylime_registrar_service_port>
+    export DIB_KEYLIME_AGENT_PORT=<keylime_agent_port> (default to 8890)
+    disk-image-create centos baremetal dhcp-all-interfaces grub2 keylime-agent -o <image_name>
+
+* TPM-emulator could be used instead of a real TPM hardware:
+
+  .. prompt:: bash $
+
+    disk-image-create centos baremetal dhcp-all-interfaces grub2 keylime-agent tpm-emulator -o <image_name>
+
+  Note: Keylime does not recommend TPM emulators for production systems and TPM 1.2 is no longer supported by Keylime.
+
+* Allowlist and its checksum could be extracted from the image initramfs or collected from the target node at boot time. Upload them to the machine that triggers attestation.
+
+* Deploy the target node with the provisioning image just built. At the node boot time, keylime-agent will start as a system service, generate its UUID, and register itself with the registrar service.
+
+Alternatively, `Keylime ansible role`_ could be used to deploy Keylime against a hardware TPM.
+
+After Keylime agent is set up on the target node, check the keylime-agent log to get agent UUID.
```

Comment:
What does it mean for the Keylime agent to be set up on the target node? I'd suggest detailing the commands to upload the image into glance; and then to provision the node-to-be-attested with the image.